### PR TITLE
DOC Add snapshot_user built-in role

### PR DESCRIPTION
**Pull request by albertzaharovits**
The `snapshot_user` role was added in https://github.com/elastic/elasticsearch/pull/35820 , this is the docs part. It also adds a doc snippet on the `create_snapshot` cluster privilege.

Follow-up on https://github.com/elastic/elasticsearch/pull/35820#discussion_r251148091

### CODE REVIEW

File: docs/en/stack/security/authorization/built-in-roles.asciidoc [hunk 1]
```diff
@@ -111,6 +111,14 @@ user has access to only their own reports. Reporting users should also be
 assigned the `kibana_user` role and a role that grants them access to the data 
 that will be used to generate reports.
 
+[[built-in-roles-snapshot-user]] `snapshot_user`::
+Grants the necessary privileges to create snapshots of **all** the indices and
+to view their metadata. This role enables users to view the configuration of
+existing snapshot repositories and of the snapshot details. It does not enable
```
Comment (lcawl) [inline docs/en/stack/security/authorization/built-in-roles.asciidoc]: Unless there's a "configuration of snapshot details", I think this should be changed to:

```suggestion
existing snapshot repositories and snapshot details. It does not enable
```

File: docs/en/stack/security/authorization/built-in-roles.asciidoc [hunk 2]
```diff
@@ -111,6 +111,14 @@ user has access to only their own reports. Reporting users should also be
...
+users to change or restore any of them. It also does not enable users to change
```
Comment (lcawl) [inline docs/en/stack/security/authorization/built-in-roles.asciidoc]: The "them" is a little unclear here. Is it just "snapshots", "snapshot repositories", or "snapshot details"?

Comment (albertzaharovits) [reply docs/en/stack/security/authorization/built-in-roles.asciidoc]: All of them. This is poorly worded. I tried to say a lot in a few words and it did not turn up well.
I have pushed https://github.com/elastic/stack-docs/pull/212/commits/6ba2c46d80551ee5ad8159fdddea8006dee748ac

**Comment by albertzaharovits** [comment]
Lisa suggested I open a PR for the kibana's side of cluster privileges and I followed suit https://github.com/elastic/kibana/pull/31086 . Thank you Lisa!

File: docs/en/stack/security/authorization/built-in-roles.asciidoc [hunk 3]
```diff
@@ -111,6 +111,12 @@ user has access to only their own reports. Reporting users should also be
 assigned the `kibana_user` role and a role that grants them access to the data 
 that will be used to generate reports.
 
+[[built-in-roles-snapshot-user]] `snapshot_user`::
+Grants the necessary privileges to create snapshots of **all** the indices and
+to view their metadata. This role enables users to view the configuration of
+existing snapshot repositories and snapshot details. It does not permit to
```
Comment (lcawl) [inline docs/en/stack/security/authorization/built-in-roles.asciidoc]: > It does not permit to

I'd suggest changing this to something like "enable users to" or "grant authority to"

File: docs/en/stack/security/authorization/built-in-roles.asciidoc [hunk 4]
```diff
@@ -111,6 +111,12 @@ user has access to only their own reports. Reporting users should also be
...
+remove or add repositories or to restore snapshots. It also does not permit
```
Comment (lcawl) [inline docs/en/stack/security/authorization/built-in-roles.asciidoc]: > It also does not permit

Ditto

**Pull request merged**